src/CommandBasedBackendRegisterInfo.cc

```diff
@@ -13,18 +106,180 @@ namespace ChimeraTK {
+        break;
+      default:
+      case InternalType::INT64:
+        dataDescriptor = DataDescriptor(
+            DataDescriptor::FundamentalType::numeric, /*isIntegral*/ true, /*isSigned*/ true, /*nDigits*/ 11);
+    }
+  } // end constructor
+
+  /******************************************************************************************************************/
+
+  CommandBasedBackend::MapFile CommandBasedBackend::parse(const std::string& mapFileName) {
+    std::ifstream file(mapFileName);
+    if(!file.is_open()) {
+      throw ChimeraTK::logic_error("Could not open the map file " + mapFileName);
+    }
+
+    json j;
+    file >> j;
+    return j.get<CommandBasedBackend::MapFile>();
+    // get() calls from_json(const json& j, CommandBasedBackend::MapFile& d)
+  }
+
+  /********************************************************************************************************************/
+
+  /**
+   * Throws a ChimeraTK::logic_error if the json key is not in an std::array of valid keys
+   */
+  template<size_t N>
+  void throwIfHasInvalidJsonKey(const json& j, const std::array<std::string, N>& validKeys, std::string errorMessage) {
+    for(auto it = j.begin(); it != j.end(); ++it) {
+      bool stringIsNotInArray = true;
+      for(const std::string& i : validKeys) {
+        if(it.key() == i) {
+          stringIsNotInArray = false;
+          break;
+        }
+      }
+
+      if(stringIsNotInArray) {
+        throw ChimeraTK::logic_error(errorMessage + " " + it.key());
+      }
+    }
+  }
+
+  /********************************************************************************************************************/
+
+  void from_json(const json& j, CommandBasedBackend::MapFile& d) {
+    if(!j.contains(to_str(METADATA)) || !j.contains(to_str(REGISTERS))) {
+      throw ChimeraTK::logic_error("Missing keys in JSON data");
+    }
+
+    throwIfHasInvalidJsonKey(j, topLevelKeyStrs, "Map file top level has unknown key");
+
+    d.metaData = j.at(to_str(METADATA)).template get<CommandBasedBackend::MetaData>();
+    // get() calls from_json(const json& j, CommandBasedBackend::MetaData& m)
+
+    for(auto& [key, value] : j.at(to_str(REGISTERS)).items()) {
+      throwIfHasInvalidJsonKey(value, registerKeyStrs, "Map file registry entry " + key + " has unknown key");
+
+      CommandBasedBackendRegisterInfo reg = getJsonToCommandBasedBackendRegisterInfo(value, key);
+      reg.registerPath = RegisterPath(key);
+      d.registers.push_back(reg);
+    }
+  }
+
+  /********************************************************************************************************************/
+
+  void from_json(const json& j, CommandBasedBackend::MetaData& m) {
+    if(!j.contains(to_str(MAP_FILE_FORMAT_VERSION))) {
+      throw ChimeraTK::logic_error("Missing mapFileFormatVersion key in metadata");
+    }
+
+    throwIfHasInvalidJsonKey(j, metadataKeyStrs, "Map file metadata has unknown key");
```

Comment:
I don't know if we want this for meta data. Usually meta data means list of arbitrary key/value pairs which are put into the meta data catalogue (which exists in addition to the register catalogue). We should fill this catalogue as well, and then check that the mandatory parameters are in there.